web/src/main/java/org/springframework/security/web/csrf/DefaultCsrfToken.java

```diff
@@ -72,6 +81,73 @@ public String getParameterName() {
 	 * @see org.springframework.security.web.csrf.CsrfToken#getToken()
 	 */
 	public String getToken() {
```

Comment:
I do agree that it should be new implementation. It might seemed the new implementation will be related with gh-7539 ?

It doesn't feel right because of the following issues:

1.  It seemed there is a "hidden rules" in generating CsrfToken string. it should be HTML safe, hence why I use Base64

2. Since the `getToken` use Base64 encoding, `matches` also require to decode Base64 in order to get the value.

It seemed that there should be `serialization` and `deserialization` strategy for the token string to deserialized to CsrfToken. Which might require CSRF token string to have data structure other than a simple UUID.

Reply:
I don't believe you need to implement gh-7539 with this ticket. That is more about how the token is read from the request vs what is stored. The logic is more like:

```
storeCsrfToken.matches(readCsrfToken.getValue())
```

That means only the stored value needs to be a custom implementation.